ultrack/core/export/trackmate.py

```diff
@@ -172,12 +225,13 @@ def tracks_layer_to_trackmate(
 
         ET.SubElement(filtered_tracks_elem, "TrackID").set("TRACK_ID", str(track_id))
 
-        for spot_id, entry in group.iterrows():
+        for spot_id, entry in group.iterrows():  # spot_id is the DataFrame row index
             parent_id = int(entry["parent_id"])
             if parent_id == NO_PARENT:
                 continue
             edge_elem = ET.SubElement(track_elem, "Edge")
-            edge_elem.set("SPOT_SOURCE_ID", str(parent_id))
+            parent_spot_id = tracks_df.index[tracks_df['id'] == parent_id][0]  # uniqueness checked above
```

Comment:
This line creates a bottleneck during conversion.
In one of our 3D datasets, this function call went from 1min40secs to 6min.

Reply:
Thanks @JoOkuma, I didn't think of the original design of tracking and your fix is indeed way better. I made a new commit, and I'll resolve the conversations.